go-controller/pkg/node/gateway_init.go

```diff
@@ -457,6 +457,13 @@ func (nc *DefaultNodeNetworkController) initGatewayDPUHost(kubeNodeIP net.IP) er
 		return err
 	}
 
+	// Delete stale masquerade resources if there are any. This is to make sure that there
+	// are no Linux resources with IP from old masquerade subnet when masquerade subnet
+	// gets changed as part of day2 operation.
+	if err := deleteStaleMasqueradeResources(gwIntf, nc.name, nc.watchFactory); err != nil {
+		return fmt.Errorf("failed to remove stale masquerade resources: %w", err)
```

Comment:
Just to confirm, we want to stick to a hard fail on cleanup issues?

Reply:
so the way it is written is so that an unexpected error, like failure to parse the kapi node, or unable to calculate the masquerade ips from the annotation value will result in an error. A failure to clean up the resources will only log inside of the deleteStaleMasqueradeResources

Reply:
Ah: https://github.com/ovn-org/ovn-kubernetes/pull/4496/files#diff-d3aa58d9b58a0a09264f072df46ab01d0501eb508c4656411ae2dc1ac68fb3c4R2233
thanks! that makes sense